From c92aa841a1e15311aa014853fa1e860a5c865e96 Mon Sep 17 00:00:00 2001
From: Brian Ruthven <Brian.Ruthven@oracle.com>
Date: Mon, 28 Mar 2022 17:50:49 -0700
Subject: [PATCH 010/105] x86emu: re-align breaks in ins() and outs()

Makes the 4-byte cases match those for 1- & 2-byte handling,
moving the break from being unconditionally hit the first time
through the to loop to after the loop is done.

Fixes Solaris Studio compiler warnings:
"prim_ops.c", line 2626: warning: end-of-loop code not reached
"prim_ops.c", line 2692: warning: end-of-loop code not reached

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 hw/xfree86/x86emu/prim_ops.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/x86emu/prim_ops.c b/hw/xfree86/x86emu/prim_ops.c
index 35ef94710..47877eb26 100644
--- a/hw/xfree86/x86emu/prim_ops.c
+++ b/hw/xfree86/x86emu/prim_ops.c
@@ -2622,8 +2622,8 @@ ins(int size)
                 store_data_long_abs(M.x86.R_ES, M.x86.R_DI,
                                     (*sys_inl) (M.x86.R_DX));
                 M.x86.R_DI += inc;
-                break;
             }
+            break;
         }
         M.x86.R_CX = 0;
         if (M.x86.mode & SYSMODE_PREFIX_DATA) {
@@ -2688,8 +2688,8 @@ outs(int size)
                 (*sys_outl) (M.x86.R_DX,
                              fetch_data_long_abs(M.x86.R_ES, M.x86.R_SI));
                 M.x86.R_SI += inc;
-                break;
             }
+            break;
         }
         M.x86.R_CX = 0;
         if (M.x86.mode & SYSMODE_PREFIX_DATA) {
-- 
2.45.2

